src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1785,6 +1810,42 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetupSrpHost(co
     return error;
 }
 
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_ClearSrpHost(const char * aHostName)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    Impl()->LockThreadStack();
+
+    VerifyOrExit(aHostName, error = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(mSrpClient.mInitializedCallback, CHIP_ERROR_INCORRECT_STATE);
+
+    // Add host and remove it with notifying SRP server to clean old information related to the host.
+    // Avoid adding the same host name multiple times
+    if (strcmp(mSrpClient.mHostName, aHostName) != 0)
+    {
+        strcpy(mSrpClient.mHostName, aHostName);
+        error = MapOpenThreadError(otSrpClientSetHostName(mOTInst, mSrpClient.mHostName));
+        SuccessOrExit(error);
+    }
+    error = MapOpenThreadError(otSrpClientRemoveHostAndServices(mOTInst, false, true));
+
+exit:
+    Impl()->UnlockThreadStack();
+
+    return error;
+}
+
+template <class ImplClass>
+CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetSrpDnsCallbacks(DnsAsyncReturnCallback aInitCallback,
+                                                                                    DnsAsyncReturnCallback aErrorCallback,
```

Comment:
`aErrorCallback` is unused, dont we have any warnings?

Reply:
Yes we do and I just wanted to utilize this callback for handling SRP server errors, but thought about doing it in separate PR, as I expect at least two of them soon.

Reply:
Ok thanks.

Reply:
But.. the CI won't pass right, so how you want to address it in the next PR? Or you meant some continuation of this?

Reply:
Are you sure? I think CI is not that strict, as build already passed few times and also current implementation has some methods with unused arguments in many places (e.g. https://github.com/project-chip/connectedhomeip/blob/master/src/platform/OpenThread/MdnsImpl.cpp#L29). I think it should fail in case of defining an unused variable, but not for the unused argument.